pkg/cmd/template/cmd.go

```diff
@@ -60,7 +60,7 @@ func NewCmd(o *Options) *cobra.Command {
 		RunE:    func(c *cobra.Command, args []string) error { return o.Run() },
 	}
 	cmd.Flags().BoolVar(&o.IgnoreUnknownComments, "ignore-unknown-comments", false,
-		"Configure whether unknown comments are considered as errors (comments that do not start with '#@' or '#!')")
+		"(deprecated: to explicitly disable processing, use `--file-mark <filename>:type=yaml-plain`) Configure whether unknown comments are considered as errors (comments that do not start with '#@' or '#!')")
```

Comment:
this description is a bit confusing since it almost implies that this flag is deprecated and i should use file marks, even though the main change here is that folks dont need to do anything anymore. this also applies to the --implicit-map-key-overrides flag.

i wouldnt modify this description because it's still true as is for yaml templates.

Reply:
With this detection in place, what's the need for the `--ignore-unknown-comments` flag?

Reply:
this flag would still be used to ignore unknown comments within legit ytt templates if user wants that.

Reply:
Why would they want this... like what's the use-case that says, "I authored this file, but I still want YAML comments"?

It invites failure states. What convenience does it afford?

Reply:
i can think of one case: i have a third party yaml file with bunch of comments. i want to start using ytt templating in it so i add one ytt directive. without having to fix all the other comments _right this moment_ i want to start seeing ytt do its evaluation.